finality-aleph/src/network/gossip/service.rs

```diff
@@ -250,21 +268,35 @@ impl<N: RawNetwork, AD: Data, BSD: Data> Service<N, AD, BSD> {
             Some(sender) => {
                 match sender.try_send(data) {
                     Err(e) => {
-                        // Receiver can also be dropped when thread cannot send to peer. In case receiver is dropped this entry will be removed by Event::NotificationStreamClosed
-                        // No need to remove the entry here
                         if e.is_full() {
-                            warn!(
-                                target: LOG_TARGET,
-                                "Failed sending data in block sync notification channel to peer because peer_sender receiver is full: {:?}",
-                                peer
-                            );
+                            let peer_and_protocol = (peer.clone(), Protocol::BlockSync);
```

Comment:
maybe extract the following block to a separate method, as it's very similar to the one 40 lines above?